lib/resources/docker.rb

```diff
@@ -0,0 +1,341 @@
+# encoding: utf-8
+#
+# Copyright 2017, Christoph Hartmann
+#
+# author: Christoph Hartmann
+# author: Patrick Muench
+# author: Dominik Richter
+
+require 'utils/filter'
+require 'hashie/mash'
+
+class DockerContainerFilter
+  # use filtertable for containers
+  filter = FilterTable.create
+  filter.add_accessor(:where)
+        .add_accessor(:entries)
+        .add(:commands,       field: 'command')
+        .add(:ids,            field: 'id')
+        .add(:images,         field: 'image')
+        .add(:labels,         field: 'labels')
+        .add(:local_volumes,  field: 'localvolumes')
+        .add(:mounts,         field: 'mounts')
+        .add(:names,          field: 'names')
+        .add(:networks,       field: 'networks')
+        .add(:ports,          field: 'ports')
+        .add(:running_for,    field: 'runningfor')
+        .add(:sizes,          field: 'size')
+        .add(:status,         field: 'status')
+        .add(:exists?) { |x| !x.entries.empty? }
```

Comment:
This feels like we need a default `exists?` accessor in `FilterTable` that can be added with `.add_accessor`. Obviously don't address this as part of this PR, but I've seen this multiple times so it feels ripe for consolidation.